src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.State.cs

```diff
@@ -110,6 +110,8 @@ internal sealed class State
             /// The <see cref="DateTime"/> values here do not update dynamically.
             /// </remarks>
             public IImmutableDictionary<string, DateTime> AdditionalDependentFileTimes { get; }
+            private ImmutableHashSet<string> LastAdditionalDependentFileTimes { get; }
```

Comment:
I'm pretty sure you don't need this field. Just use `AdditionalDependentFileTimes` instead. When updating the state, the last times are the ones of the old snapshot, the new ones are available. Of they've changed, update `LastAdditionalDependentFileTimesChangedAtUtc`.